Labs 74 mysql auto subscriber provider

vokoblin opened the pull request:
Getting all the existent subscribers.

Commit:
Creating AutoSubscriberProvider MysqlGateway:
2 basic tests and empty sql file.

Commit:
adds database subscribers table

Commit:
Adds all tests and creates User Entity

Commit:
Adding a test for getId();
Refactoring code.
Formatting code.

MeliaBurnley commented on src/AutoSubscribe/Entity/User.php:
```diff
@@ -0,0 +1,37 @@
+    /**
+     * @var
+     */
+    private $userId;
```
Type

MeliaBurnley commented on test/AutoSubscribe/MysqlAutoSubscriberProviderTest.php:
```diff
@@ -0,0 +1,94 @@
+    public function duringConstructionOfGateway_GatewayIsNotNull()
+    {
+        $this->assertNotNull( $this->gateway );
+    }
```
I'm a little confused as to why you need this?

MeliaBurnley commented on test/AutoSubscribe/MysqlAutoSubscriberProviderTest.php:
```diff
@@ -0,0 +1,94 @@
+    /**
+     * @test
+     */
+    public function givenNoSuscribers_ReturnEmptyArray()
```
"Suscribers" -> "Subscribers"

MeliaBurnley commented on test/AutoSubscribe/MysqlAutoSubscriberProviderTest.php:
```diff
@@ -0,0 +1,94 @@
+        $subscriberName = "TheSubscribes";
+        $subscriber = new User( $subscriberName );
+        $this->assertEquals( $subscriberName, $subscriber->getId() );
+    }
```
Also unsure as to what this is for, surely that's testing User and should be in a separate test?

Commit:
Makes changes:
Creates UserTest;
Removes UserTest from MysqlAutoSubscriberProviderTest;
Removes un-needed test;
Adds types of the variables in User;
It was decided taht User id should be String but in the interface its a
INT.

MeliaBurnley commented on db/3.sql:
```diff
@@ -0,0 +1,5 @@
+CREATE TABLE `subscribers` (
```
You're gonna want to stick this in the labs-schema repo now

danielburnley commented on src/AutoSubscribe/Entity/User.php:
```diff
@@ -0,0 +1,37 @@
+ * Date: 03/09/2015
+ * Time: 12:16
+ */
+class User implements IUser
```
Is there a reason we have so many user classes that all do the exact same things?

MeliaBurnley commented on test/AutoSubscribe/Entity/UserTest.php:
```diff
@@ -0,0 +1,57 @@
+     * @param User $user
+     * @param array $emptyUserData
+     */
+    private function assertUserGettersMatchExpectedUserData( User $user, $emptyUserData )
```
Should be up above setUp

Commit:
Merge branch 'master' of github.com:clearbooks/labs-mysql into LABS-74_MysqlAutoSubscriberProvider

Commit:
Removing un-used db directory

Commit:
Updating composer

Commit:
Moving it above the setUp()
